sdk/src/main/scala/com/horizen/account/api/http/AccountTransactionApiRoute.scala

```diff
@@ -254,14 +261,235 @@ case class AccountTransactionApiRoute(override val settings: RESTApiSettings,
+        }
+
+      }
+    }
+  }
+
+  def allWithdrawalRequests: Route = (post & path("allWithdrawalRequests")) {
+    entity(as[ReqAllWithdrawalRequests]) { body =>
+      withNodeView { sidechainNodeView =>
+        val accountState = sidechainNodeView.getNodeState
+        val listOfWithdrawalRequests = accountState.withdrawalRequests(body.epochNum)
+        ApiResponseUtil.toResponse(RespAllWithdrawalRequests(listOfWithdrawalRequests.toList))
+      }
+    }
+  }
+
+  def createSmartContract: Route = (post & path("createSmartContract")) {
+    entity(as[ReqCreateContract]) { body =>
+      // lock the view and try to create CoreTransaction
+      applyOnNodeView { sidechainNodeView =>
+        val valueInWei = BigInteger.ZERO
+        // TODO actual gas implementation
+        var maxFeePerGas = BigInteger.ONE
+        var maxPriorityFeePerGas = BigInteger.ONE
+        var gasLimit = BigInteger.ONE
+        if (body.gasInfo.isDefined) {
+          maxFeePerGas = body.gasInfo.get.maxFeePerGas
+          maxPriorityFeePerGas = body.gasInfo.get.maxPriorityFeePerGas
+          gasLimit = body.gasInfo.get.gasLimit
+        }
+        //TODO Probably getFittingSecret would need to take into account also gas
+        val secret = getFittingSecret(sidechainNodeView, None, valueInWei)
+        secret match {
+          case Some(secret) =>
+            val to = null
+            val nonce = body.nonce.getOrElse(sidechainNodeView.getNodeState.getNonce(secret.publicImage.address))
+            val data = body.contractCode
+            val tmpTx: EthereumTransaction = new EthereumTransaction(
+              body.chainId,
+              to,
+              nonce,
+              gasLimit,
+              maxPriorityFeePerGas,
+              maxFeePerGas,
+              valueInWei,
+              data,
+              null
+            )
+            validateAndSendTransaction(signTransactionWithSecret(secret, tmpTx))
+          case None =>
+            ApiResponseUtil.toResponse(ErrorInsufficientBalance("No account with enough balance found", JOptional.empty()))
+        }
+
+      }
+    }
+  }
+
+
+  def encodeAddNewStakeCmdRequest(forgerStakeInfo: TransactionForgerOutput): String = {
+    val blockSignPublicKey = new PublicKey25519Proposition(BytesUtils.fromHexString(forgerStakeInfo.blockSignPublicKey.getOrElse(forgerStakeInfo.ownerAddress)))
+    val vrfPubKey = new VrfPublicKey(BytesUtils.fromHexString(forgerStakeInfo.vrfPubKey))
+    val addForgerStakeInput = AddNewStakeCmdInput(ForgerPublicKeys(blockSignPublicKey, vrfPubKey), new AddressProposition(BytesUtils.fromHexString(forgerStakeInfo.ownerAddress)))
+    val data = BytesUtils.toHexString(Bytes.concat(BytesUtils.fromHexString(ForgerStakeMsgProcessor.AddNewStakeCmd), addForgerStakeInput.encode()))
+    data
+  }
+
+  def encodeSpendStakeCmdRequest(spendForgerStake: TransactionSpendForgerStake): String = {
+    val signature = new SignatureSecp256k1(BytesUtils.fromHexString(spendForgerStake.v),
```

Comment:
We don't expect to retrieve the signature from the user.
User give us only `stakeId`. Then we shoudl take the owner from the State, and create a signature using the corresponding private key from the wallet.

Reply:
Done